TestNetLongName fails on kvm

**Environment**
Semaphore, testing the code on master, after kvm tests has been enabled with https://github.com/coreos/rkt/pull/2007.

**What did you expect to see?**
Tests pass on master.

**What did you see instead?**

```
--- FAIL: TestNetLongName (2.65s)
	rkt_net_test.go:620: Telling the child to connect via http://192.168.122.248:49152
	goroutineassistant.go:76: Command: /home/runner/rkt/builds/build-rkt-kvm/build-rkt-1.4.0+git/tmp/functional/rkt --dir=/tmp/datadir-515443606 --local-config=/tmp/localdir-194199037 --system-config=/tmp/systemdir-711369016 --debug --insecure-options=image run --net=thisnameiswaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaytoolong --mds-register=false /home/runner/rkt/builds/build-rkt-kvm/build-rkt-1.4.0+git/tmp/functional/test-tmp/rkt-inspect-networking.aci
	goroutineassistant.go:64: Error encountered - shutting down
	goroutineassistant.go:65: Error: ExpectRegex didn't find regex 'HTTP-Get received: (.*?)\r'.
		Output: image: using image from file /home/runner/rkt/builds/build-rkt-kvm/build-rkt-1.4.0+git/tmp/functional/stage1-kvm.aci

		image: using image from file /home/runner/rkt/builds/build-rkt-kvm/build-rkt-1.4.0+git/tmp/functional/test-tmp/rkt-inspect-networking.aci

		stage0: Preparing stage1

		stage0: Writing image manifest

		stage0: Loading image sha512-b90901ef8129b1d5595eae4717b234e41ee8add5d678e8a06bb7d47ff5fcaac1

		stage0: Writing image manifest

		stage0: Writing pod manifest

		stage0: Setting up stage1

		stage0: Wrote filesystem to /tmp/datadir-515443606/pods/run/4a8856dd-a1c6-4837-95e0-6b5075aef32a

		stage0: Pivoting to filesystem /tmp/datadir-515443606/pods/run/4a8856dd-a1c6-4837-95e0-6b5075aef32a

		stage0: Execing /init

		networking: loading networks from /tmp/localdir-194199037/net.d

		stage1: failed to setup network

		  └─exit status 2: iptables v1.4.21: chain name `CNI-thisnameiswaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaytoolong-6dcb90730a39298e' too long (must be under 29 chars)

		Try `iptables -h' or 'iptables --help' for more information.		
```

--- a/networking/kvm.go
+++ b/networking/kvm.go
@@ -29,6 +29,7 @@ import (
 
 	"github.com/appc/cni/pkg/ip"
 	cnitypes "github.com/appc/cni/pkg/types"
+	cniutils "github.com/appc/cni/pkg/utils"
 	"github.com/appc/spec/schema/types"
 	"github.com/hashicorp/errwrap"
 	"github.com/vishvananda/netlink"
@@ -42,7 +43,6 @@ const (
 	defaultBrName     = "kvm-cni0"
 	defaultSubnetFile = "/run/flannel/subnet.env"
 	defaultMTU        = 1500
-	masqComment       = "rkt-lkvm masquerading"
 )
 
 type BridgeNetConf struct {
@@ -541,11 +541,12 @@ func kvmSetup(podRoot string, podID types.UUID, fps []ForwardedPort, netList com
 		}
 
 		if n.conf.IPMasq {
-			chain := getChainName(podID.String(), n.conf.Name)
+			chain := cniutils.FormatChainName(n.conf.Name, podID.String())
+			comment := cniutils.FormatChainName(n.conf.Name, podID.String())
 			if err := ip.SetupIPMasq(&net.IPNet{
 				IP:   n.runtime.IP,
 				Mask: net.IPMask(n.runtime.Mask),
-			}, chain, masqComment); err != nil {
+			}, chain, comment); err != nil {
 				return nil, err
 			}
 		}
@@ -598,11 +599,12 @@ func (n *Networking) teardownKvmNets() {
 		}
 		// remove masquerading if it was prepared
 		if an.conf.IPMasq {
-			chain := getChainName(n.podID.String(), an.conf.Name)
+			chain := cniutils.FormatChainName(an.conf.Name, n.podID.String())
+			comment := cniutils.FormatChainName(an.conf.Name, n.podID.String())
 			err := ip.TeardownIPMasq(&net.IPNet{
 				IP:   an.runtime.IP,
 				Mask: net.IPMask(an.runtime.Mask),
-			}, chain, masqComment)
+			}, chain, comment)
 			if err != nil {
 				stderr.PrintE("error on removing masquerading", err)
 			}

--- a/tests/rkt_net_nspawn_test.go
+++ b/tests/rkt_net_nspawn_test.go
@@ -57,20 +57,3 @@ func TestNetCustomPtp(t *testing.T) {
 func TestNetDefaultConnectivity(t *testing.T) {
 	NewNetDefaultConnectivityTest().Execute(t)
 }
-
-// TODO: fix this for kvm, see https://github.com/coreos/rkt/issues/2533
-func TestNetLongName(t *testing.T) {
-	nt := networkTemplateT{
-		Name:   "thisnameiswaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaytoolong",
-		Type:   "ptp",
-		IpMasq: true,
-		Ipam: ipamTemplateT{
-			Type:   "host-local",
-			Subnet: "11.11.6.0/24",
-			Routes: []map[string]string{
-				{"dst": "0.0.0.0/0"},
-			},
-		},
-	}
-	testNetCustomNatConnectivity(t, nt)
-}

--- a/tests/rkt_net_test.go
+++ b/tests/rkt_net_test.go
@@ -783,3 +783,19 @@ func NewNetOverrideTest() testutils.Test {
 		}
 	})
 }
+
+func TestNetLongName(t *testing.T) {
+	nt := networkTemplateT{
+		Name:   "thisnameiswaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaytoolong",
+		Type:   "ptp",
+		IpMasq: true,
+		Ipam: ipamTemplateT{
+			Type:   "host-local",
+			Subnet: "11.11.6.0/24",
+			Routes: []map[string]string{
+				{"dst": "0.0.0.0/0"},
+			},
+		},
+	}
+	testNetCustomNatConnectivity(t, nt)
+}

